Tasks/Common/codecoverage-tools/cobertura/cobertura.maven.ccenabler.ts

```diff
@@ -0,0 +1,138 @@
+            .thenResolve(true);
+    }
+
+    protected applyFilterPattern(filter: string): string[] {
+        let ccfilter = [];
+        let _this = this;
+
+        if (!util.isNullOrWhitespace(filter)) {
+            str(util.trimToEmptyString(filter)).replaceAll(".", "/").s.split(":").forEach(exFilter => {
+                if (exFilter) {
+                    ccfilter.push(str(exFilter).endsWith("*") ? (exFilter + "/**") : (exFilter + ".class"));
+                }
+            });
+        }
+
+        return ccfilter;
+    }
+
+    protected addCodeCoverageNodes(buildJsonContent: any): Q.Promise<any> {
+        let _this = this;
+        let pluginsNode = null;
+        let isMultiModule = false;
+        let buildNode = null;
+        
+        if (!buildJsonContent.project) {
+            return Q.reject("Invalid build file");
+        }
+
+        if (buildJsonContent.project.modules) {
+            console.log("Multimodule project detected");
+            isMultiModule = true;
+        }
+
+        if (!buildJsonContent.project.build) {
+            console.log("Build tag is not present");
+            buildJsonContent.project.build = {};
+        }
+
+        if (!buildJsonContent.project.build || typeof buildJsonContent.project.build === "string") {
+            buildNode = {};
+            buildJsonContent.project.build = buildNode;
+        }
+
+        if (buildJsonContent.project.build instanceof Array) {
+            if(typeof buildJsonContent.project.build[0] === 'string'){
+                buildNode = {};
+                buildJsonContent.project.build[0] = buildNode;
+            }else{
+                buildNode = buildJsonContent.project.build[0];
+            }
+        }
+
+        if (buildNode.pluginManagement) {
+            if (typeof buildNode.pluginManagement === "string") {
+                buildNode.pluginManagement = {};
+            }
+            if (buildNode.pluginManagement instanceof Array) {
+                pluginsNode = buildNode.pluginManagement[0].plugins;
+            }
+            else {
+                pluginsNode = buildNode.pluginManagement.plugins;
+            }
+        }
+        else {
+            if (!buildNode.plugins || typeof buildNode.plugins === "string") {
+                buildNode.plugins = {};
+            }
+            if (buildNode.plugins instanceof Array) {
+                if(typeof buildNode.plugins[0] === 'string'){
+                    pluginsNode = {};
+                    buildNode.plugins[0] = pluginsNode;
+                }
+                else{
+                    pluginsNode = buildNode.plugins[0];
+                }
+            }else{
+                pluginsNode = buildNode.plugins;
+            }
+        }
+
+        let ccPluginData = ccc.coberturaMavenEnable(_this.includeFilter, _this.excludeFilter, String(isMultiModule));
+        let reportContent = ccc.coberturaMavenReport();
+
+        return Q.allSettled([ccPluginData, reportContent])
```

Comment:
Would it be possible to configure Maven to run the cobertura plugin directly instead of modifiying the pom.xml ? For SonarQube and PMD we managed to add sonar:sonar / pmd:pmd goals to the plugins and use the command line to pass various parameters. Granted, for PMD not all of the configration was possible via the command line.